neutronics/mcnp_inputs.py

```diff
@@ -0,0 +1,157 @@
+import math
+from string import Template
+import material_data as md
+
+def merge_comps(compA, compB):
+    """Merge two compositions for homogenization. Combine like isotopes when
+    necessary. This function takes two input compositions, and returns the union
+    of the two compositions.
+
+    Arguments:
+    ----------
+        compA (dict): first composition
+        compB (dict): second composition
+
+    Returns:
+    --------
+        compA (dict): merged comp. of A and B
+    """
+    for isotope in compB:
+        if isotope in compA:
+            compA[isotope] += compB[isotope]
+        else:
+            compA.update({isotope : compB[isotope]})
+
+    return compA
+
+
+class HomogeneousInput:
+    """Write Homogeneous Input File.
+    Class to write homogeneous MCNP burnup input files.
+    """
+    
+    def __init__(self, radius, volfrac_fuel, length):
+        """Initialize geometric reactor parameters.
+
+        Initialized Attributes:
+        -----------------------
+            z (float): reactor core height
+            r (float): reactor core radius
+            frac_fuel (float): fuel to coolant channel fraction
+        """
+        self.z = length
+        self.r = radius
+        self.frac_fuel = volfrac_fuel
+        self.refl_t = 15 # 15 cm reflector
+
+    def calculate_volume(self):
+        """Get volumes for core and reflector regions.
+        
+        Modified Attributes:
+        --------------------
+            vol (float): homogenized core volume
+            vol_refl (float): reflector volume
+        """
+        self.core_vol = self.r**2 * math.pi * self.z
+        self.refl_vol = ((self.r + self.refl_t)**2 - self.r**2)*math.pi * self.z
+
+    
+    def homog_core(self, enrich=0.9):
+        """Homogenize the fuel, clad, and coolant.
+        
+        Arguments:
+        ----------
+            enrich (float) (optional): uranium enrichment
+        
+        Modified Attributes:
+        --------------------
+            rho (float): fuel density
+            fuel_string (str): mcnp-style fuel string
+        """
+        ############ Channel Dimensions ################
+        # Radius = 0.5 cm, clad_thick = 0.0031         #
+        clad_to_cool = 0.127844                        #
```

Comment:
It might be cleaner to calculate this from the radius & thickness - makes it easier to change if/when needed.